File: tony/java/TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            Integer diff = target - nums[i];
            if (map.containsKey(diff)) {
                int[] indices = {map.get(diff), i};
                return indices;
            }

            map.put(nums[i], i);
        }

        return null;
    }
}

File: lulu/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash_dict = {}
        for index, x in enumerate(nums):
            #print(hash_dict)
            if x in hash_dict:
                return [hash_dict[x], index]
            else:
                hash_dict[target - x] = index


if __name__ == '__main__':
    nums = [2,7,11,15]
    target = 9
    mySolution = Solution()
    print(mySolution.twoSum(nums, target))

            
                

        
